nexus/reconfigurator/planning/src/planner.rs

```diff
@@ -896,6 +895,126 @@ impl<'a> Planner<'a> {
         Ok(())
     }
 
+    /// Update at most one existing zone to use a new image source.
+    fn do_plan_zone_updates(&mut self) -> Result<(), Error> {
+        // We are only interested in non-decommissioned sleds.
+        let sleds = self
+            .input
+            .all_sleds(SledFilter::Commissioned)
+            .map(|(id, _details)| id)
+            .collect::<Vec<_>>();
+
+        // Wait for zones to appear up-to-date in the inventory.
+        let inventory_zones = self
+            .inventory
+            .all_ledgered_omicron_zones()
```

Comment:
This is correct for `main`, because "the ledgered zones" and "the running zones" are identical. But I wanted to ask whether I should change this on #8064. As of #8064, there will be a meaningful difference between "the ledgered zones" and "the zones actually running" - ledgering happens first, then there's a delay (usually short, but not guaranteed), then the running zones are updated later. Should this check be "all sleds know they should be running the new image sources" or "all sleds are actually running the new image sources"? Probably the latter?